neon_audio/tts/neon.py

```diff
@@ -134,19 +132,32 @@ def _parse_message(self, message, **kwargs):
         sentence = message.data["text"]
         responses = {}
         for request in tts_requested:
-            lang = request["language"]
-            if lang in self.cached_translations:
-                sentence = self.cached_translations[lang].get(sentence)
-            elif lang.split("-")[0] != "en":  # TODO: Internal lang DM
-                self.cached_translations[lang][sentence] = self.translator.translate(sentence, lang, "en")
-                self.cached_translations.store()
-                sentence = self.cached_translations[lang][sentence]
-                request["translated"] = True
+            lang = kwargs["lang"] = request["language"]
             wav_file, phonemes = self._get_tts(sentence, **kwargs)
             if not responses.get(lang):
-                responses[lang] = {"sentence": sentence}
-            if os.path.isfile(wav_file):  # Based on <speak> tags, this may not exist
-                responses[lang][request["gender"]] = wav_file
+                responses[lang] = {"sentence": sentence,
+                                   "translated": False,
+                                   "phonemes": phonemes,
+                                   request["gender"]: wav_file}
+            # translate if needed
+            if lang.split("-")[0] != self.lang.split("-")[0]:
+                tx_sentence = None
+                if lang in self.cached_translations:
+                    tx_sentence = self.cached_translations[lang].get(sentence)
+                else:
+                    self.cached_translations[lang] = {}
+                if not tx_sentence:
+                    tx_sentence = self.translator.translate(sentence, lang, self.lang)
+                    self.cached_translations[lang][sentence] = tx_sentence
+                    self.cached_translations.store()
+                tx_kwargs = dict(kwargs)
+                tx_kwargs["lang"] = self.lang
+                wav_file, phonemes = self._get_tts(tx_sentence, **tx_kwargs)
+                responses[self.lang] = {"sentence": tx_sentence,
```

Comment:
This looks like `responses['lang']`, not `self.lang`

Reply:
same answer as above, both langs are included